lib/post_creator.rb

```diff
@@ -362,13 +362,15 @@ def find_category_id
   end
 
   def create_topic
-    return if @topic
-    begin
-      topic_creator = TopicCreator.new(@user, guardian, @opts)
-      @topic = topic_creator.create
-    rescue ActiveRecord::Rollback
-      rollback_from_errors!(topic_creator)
+    unless @topic
```

Comment:
This refactor here is a cosmetic change since I don't think it changes any existing logic.